tests/suites/test_suite_constant_time.function

```diff
@@ -77,13 +77,13 @@ void mbedtls_ct_bool_xxx(char *x_str, char *y_str)
     expected = x1 <= y1 ? MBEDTLS_CT_TRUE : MBEDTLS_CT_FALSE;
     TEST_EQUAL(mbedtls_ct_bool_le(x, y), expected);
 
-    expected = mbedtls_ct_bool(x) ^ mbedtls_ct_bool(y) ? MBEDTLS_CT_TRUE : MBEDTLS_CT_FALSE;
+    expected = mbedtls_ct_bool(x1) ^ mbedtls_ct_bool(y1) ? MBEDTLS_CT_TRUE : MBEDTLS_CT_FALSE;
```

Comment:
This is a bit in-between opaque and non-opaque use of `mbedtls_ct_condition_t`. We need internal knowledge in order to know that the bitwise XOR will work, and if we have this internal knowledge we also know we don't need the ` ? MBEDTLS_CT_TRUE : MBEDTLS_CT_FALSE` part. In addition to the inconsistency, I'd rather not rely on internal knowledge at all if we can avoid it.

Same for `_and` or `_or` below, where I'd recommend using logical and / or: `expected = x1 && y1 ? MBEDTLS_CT_TRUE : MBEDTLS_CT_FALSE`. Since C has no logical XOR, since would be a bit more verbose here: `(x1 && !y1) || (!x1 && y1)` (or anything equivalent).